cli/azd/pkg/input/console.go

```diff
@@ -36,15 +37,26 @@ type ConsoleOptions struct {
 	DefaultValue any
 }
 
-func (c *AskerConsole) Message(ctx context.Context, message string) error {
-	_, err := fmt.Println(message)
-	if err != nil {
-		return fmt.Errorf("error printing line: %w", err)
+// Sets the underlying writer for the console
+func (c *AskerConsole) SetWriter(writer io.Writer) {
+	if writer == nil {
+		writer = colorable.NewColorableStdout()
 	}
 
-	return nil
+	c.writer = writer
 }
 
+// Prints out a message to the underlying console write
+func (c *AskerConsole) Message(ctx context.Context, message string) {
+	// Only write to the console during interactive & non-formatted responses.
+	if c.interactive && c.formatter.Kind() == output.NoneFormat {
+		fmt.Fprintln(c.writer, message)
+	} else {
+		log.Println(message)
```

Comment:
We set this up very early in `main.go`.  If `--debug` is passed to the CLI, then these messages are written to stderr.  Otherwise they are dropped on the floor.

https://github.com/Azure/azure-dev/blob/e86f92a04116eafa5e7fc092e6db4a42687bbe49/cli/azd/main.go#L31-L35

Reply:
Prints to the standard go logger.  Docs available @ https://pkg.go.dev/log if you are curious.